packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -145,6 +179,17 @@ class SharedPreferences {
 
   /// Completes with true once the user preferences for the app has been cleared.
   Future<bool> clear() {
+    if (_prefixHasBeenChanged) {
+      try {
+        _preferenceCache.clear();
+        return _store.clearWithPrefix(_prefix);
+      } catch (e) {
```

Comment:
This should only catch `UnimplementedError` (with an ignore if that's a lint warning, which it probably is), and a comment explaining that it's catching an error for the purpose of giving it a clearer error message rather than actually handling it.

As written, this would be very confusing if an implementation threw a `PlatformException` during the clear for some reason, because we'd lose that exception and replace it with a very misleading error.